pip/pip-310.md

```diff
@@ -0,0 +1,121 @@
+# PIP-310 Support custom publish rate limiters
+
+# Background knowledge
```

Comment:
* Seems reasonable to include an explanation what are Resource Groups, where are they configured and what does it mean to apply a rate limit for them (is it local to broker or a distributed one). You won't find this info anywhere, and I bet you won't find many core Pulsar developers familiar with this
* You haven't mentioned if rate limited are server or client based. I would start with that scope in the beginning.

Reply:
I agree with you that resource groups aren't a well known feature.. But adding details about resource group (and other additional contextual details) - would that make the Background knowledge of this PIP too big?

Rate limiters in Pulsar are all server based AFAIK, please correct me if I am mistaken.

Reply:
I think the size is not a consideration and I don't think an explanation would exceed a normal length. It's the cause that matters: One can not understand a suggestion without proper background most of us don't have, right?

Reply:
Noted.